src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+  // if there is no user with a given userId
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxTodosId = [...todos].sort((a, b) => b.id - a.id)[0].id;
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isValidTitle, setIsValidTitle] = useState(true);
+  const [isValidUser, setIsValidUser] = useState(true);
+
+  const makeNewTodo = () => {
+    const newTodo = {
+      id: maxTodosId + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUser(userId),
+    };
+
+    todos.push(newTodo);
+  };
+
+  const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsValidTitle(true);
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setIsValidUser(true);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setIsValidUser(false);
+    }
+
+    if (!title.trim()) {
+      setIsValidTitle(false);
+    }
+
+    if (!userId || !title.trim()) {
+      return;
+    }
+
+    makeNewTodo();
+
+    setUserId(0);
+    setTitle('');
+    setIsValidTitle(true);
+    setIsValidUser(true);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={handleChangeInput}
+          />
+          {isValidTitle || <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <select
+            data-cy="userSelect"
+            defaultValue={userId}
```

Comment:
You could remove this ```defaultValue``` and just use as default `0` such as you set `0` as default id in 26 line
```
const [userId, setUserId] = useState(0);
```